### PR TITLE
edit unittest so that it uses GOOGL instead of GOOG

### PR DESCRIPTION
**Pull request by jihoon796**
The unit test was edited to account for the stock split on April 2, 2014. The class of shares trading on or before April 2 (known as Class A), changed its trading symbol from GOOG to GOOGL. When the markets opened on April 3, these shares of stock began trading under the new, 5-letter symbol.

Source: https://support.sigfig.com/hc/en-us/articles/202586324-What-happened-during-the-Google-Stock-split-

### CODE REVIEW

**Comment by jihoon796** [comment]
Added datetime functionality so that the get_historical_prices function also accepts datetime objects as arguments. It is especially useful when using the ystockquote module with many dates. 

The only addition to the list of dependencies is the datetime module.

File: ystockquote.py [hunk 1]
```diff
@@ -13,488 +13,497 @@
...
+	"""
+	Get historical prices for the given ticker symbol.
+	Date format is 'YYYY-MM-DD'. Also accepts datetime
+	in addition to strings.
+
+	Returns a nested dictionary (dict of dicts).
+	outer dict keys are dates ('YYYY-MM-DD')
+	"""
+
+	if isinstance(start_date, datetime.date):
```
Comment (cgoldberg) [inline ystockquote.py]: im kinda not crazy about type checking here.  is there a more pythonic way to do this with duck typing and exceptions?

**Comment by cgoldberg** [comment]
i'm not sure what the whitespace changes are in the diff?   could you revert those and keep things PEP8ish.  thanks!

File: test_ystockquote.py [hunk 2]
```diff
@@ -33,7 +33,7 @@ def test_pep8_conformance(self):
 class YStockQuoteTestCase(TestWithScenarios):
 
     def test_get_all(self):
-        symbol = 'GOOG'
+        symbol = 'GOOGL'
```
Comment (cgoldberg) [inline test_ystockquote.py]: Good call on updating the ticker symbol.

File: ystockquote.py [hunk 3]
```diff
@@ -13,488 +13,497 @@
...
+	import datetime
 except ImportError:
-    # py2
-    from urllib2 import Request, urlopen
-    from urllib import urlencode
-
+	# py2
+	from urllib2 import Request, urlopen
+	from urllib import urlencode
+	import datetime
```
Comment (cgoldberg) [inline ystockquote.py]: move this up so imports are alphabetized.

I think the correct order there should be:

```python
from urllib import urlencode
from urllib2 import Request, urlopen
import datetime
```

File: ystockquote.py [hunk 4]
```diff
@@ -13,488 +13,497 @@
...
+		day_data = day.split(',')
+		date = day_data[0]
+		hist_dict[date] = \
+			{keys[1]: day_data[1],
+			 keys[2]: day_data[2],
+			 keys[3]: day_data[3],
+			 keys[4]: day_data[4],
+			 keys[5]: day_data[5],
+			 keys[6]: day_data[6]}
+	return hist_dict
```
Comment (cgoldberg) [inline ystockquote.py]: missing newline

**Comment by cgoldberg** [comment]
Thanks for the PR!  I just reviewd and mostly just some nitpicks.. but I'm happy to merge something once it gets cleaned up.  let me know if anything wasn't clear or you'd like me to make any of the suggested changes to get this landed.

cheers.
-Corey

**Comment by jihoon796** [comment]
Thanks for your feedback, Corey! 

I've made use of duck typing/exceptions and changed back the spacing to PEP8 conventions (was using tabs instead of spaces).

**Comment by jihoon796** [comment]
Hi Corey,

Happy holidays! Thanks for looking at the PR and for your suggestions - and
thanks for your feedback on the duck typing/exceptions part.

Also, I had accidentally converted to tabs instead of spaces somewhere, so
good call on that as well.

I've made a brand new PR, so let me know if anything else catches your eye.

Cheers,
Jihoon

**Comment by cgoldberg** [comment]
oh.. in the future, you can just push more fixes to your existing PR when rework something.. no need to create a new PR unless you are separating the PR into several discrete pieces.  Anyway,, we'll followup on #40 and I'll close this one.

**Pull request closed without merging**